src/shared/Microsoft.Git.CredentialManager/CommandContext.cs

```diff
@@ -132,33 +134,39 @@ public ITerminal Terminal
 
         public ICredentialStore CredentialStore { get; } = CreateCredentialStore();
 
-        public IReadOnlyDictionary<string, string> GetEnvironmentVariables()
+        public IReadOnlyDictionary<string, string> EnvironmentVariables
         {
-            IDictionary variables = Environment.GetEnvironmentVariables();
+            get
+            {
+                if (_envars is null)
+                {
+                    IDictionary variables = Environment.GetEnvironmentVariables();
 
-            // On Windows it is technically possible to get env vars which differ only by case
-            // even though the general assumption is that they are case insensitive on Windows.
-            // For example, some of the standard .NET types like System.Diagnostics.Process
-            // will fail to start a process on Windows if given duplicate environment variables.
-            // See this issue for more information: https://github.com/dotnet/corefx/issues/13146
+                    // On Windows it is technically possible to get env vars which differ only by case
+                    // even though the general assumption is that they are case insensitive on Windows.
+                    // For example, some of the standard .NET types like System.Diagnostics.Process
+                    // will fail to start a process on Windows if given duplicate environment variables.
+                    // See this issue for more information: https://github.com/dotnet/corefx/issues/13146
 
-            // If we're on the Windows platform we should de-duplicate by setting the string
-            // comparer to OrdinalIgnoreCase.
-            var comparer = PlatformUtils.IsWindows()
-                         ? StringComparer.OrdinalIgnoreCase
-                         : StringComparer.Ordinal;
+                    // If we're on the Windows platform we should de-duplicate by setting the string
+                    // comparer to OrdinalIgnoreCase.
+                    var comparer = PlatformUtils.IsWindows()
+                        ? StringComparer.OrdinalIgnoreCase
+                        : StringComparer.Ordinal;
 
-            var result = new Dictionary<string, string>(comparer);
+                    _envars = new Dictionary<string, string>(comparer);
 
-            foreach (var key in variables.Keys)
-            {
-                if (key is string name && variables[key] is string value)
-                {
-                    result[name] = value;
+                    foreach (var key in variables.Keys)
+                    {
+                        if (key is string name && variables[key] is string value)
+                        {
+                            _envars[name] = value;
+                        }
+                    }
                 }
-            }
 
-            return result;
+                return new ReadOnlyDictionary<string, string>(_envars);
```

Comment:
You seem to be populating a new `ReadOnlyDictionary` with every call, even though `_envars` is not changing. Do you want to cache the `ReadOnlyDictionary` instead?

Reply:
Good point! Fixed.